sideload.go

```diff
@@ -0,0 +1,420 @@
+
+	// filterCheckptHeights is the slice of hardcoded filter checkpoint heights.
+	filterCheckptHeights []uint32
+
+	// blkHdrCurHeight is the height of the block header's store tip.
+	blkHdrCurHeight int32
+
+	// fHdrCurHeight is the height of the filter header's store tip.
+	fHdrCurHeight int32
+
+	// fHdrCurHash is the filter hash of the filter header's store tip.
+	fHdrCurHash *chainhash.Hash
+
+	// fHdrNextCheckptHeight is the height of the next filter header next
+	// checkpoint's height relative to the current filter header tip.
+	fHdrNextCheckptHeight int32
+
+	// fHdrLastHeight is the last height to which we can sideload the filter
+	// headers to. This restriction occurs because the filter header tip
+	// cannot surpass the block header tip as it depends on it.
+	fHdrLastHeight uint32
+}
+
+// processSideloadedCfHeader verifies and stores the passed cfheaders.
+// It returns the current height and next checkpoint after this process to be
+// used by the called. It returns an error where necessary.
+func (h *sideloader) processSideloadedCfHeader(
+	headers []*chainhash.Hash) (*chaindataloader.ProcessHdrResp, error) {
+
+	msgCfHeader := &wire.MsgCFHeaders{
+		FilterType:       fType,
+		StopHash:         *headers[len(headers)-1],
+		PrevFilterHeader: *h.fHdrCurHash,
+		FilterHashes:     headers,
+	}
+
+	// We only verify, if this is set to false.
+	if !h.fHdrSkipVerify {
+
+		if !verifyCheckpoint(h.fHdrCurHash, headers[len(headers)-1],
+			msgCfHeader) {
+
+			log.Warnf("Filter headers failed verification at height  - %v",
+				h.fHdrCurHeight)
+			return nil, nil
+		}
+		h.fHdrNextCheckpointIdx = h.fHdrNextCheckpointIdx + 1
+		h.fHdrNextCheckptHeight = int32(h.filterCheckptHeights[h.
+			fHdrNextCheckpointIdx])
+
+	}
+
+	_, _, _, err := writeCfHeaders(msgCfHeader, h.fHdrStore,
+		h.blkHdrProcessor.store)
+
+	if err != nil {
+		return nil, err
+	}
+
+	h.fHdrCurHash = headers[len(headers)-1]
+	h.fHdrCurHeight = h.fHdrCurHeight + int32(len(headers))
+
+	// Check if we have reached the height that we are not to surpass.
+	if h.fHdrCurHeight > int32(h.fHdrLastHeight) {
+		log.Debugf("Completed filter header sideloading at height %v",
+			h.fHdrLastHeight)
+		return nil, nil
+	}
+
+	return &chaindataloader.ProcessHdrResp{
+		CurHeight:         h.fHdrCurHeight,
+		NextCheckptHeight: h.fHdrNextCheckptHeight,
+	}, nil
+}
+
+// processSideLoadBlockHeader verifies and stores the passed block headers.
+// It returns the current height and next checkpoint after this process to be
+// used by the called. It returns an error where necessary.
+func (h *sideloader) processSideLoadBlockHeader(headers []*wire.
+	BlockHeader) (*chaindataloader.ProcessHdrResp, error) {
+
+	if !areHeadersConnected(headers) {
+		log.Warnf("Headers received don't connect")
+		return nil, nil
+	}
+
+	headersArray := make([]headerfs.BlockHeader, 0, len(headers))
+
+	var (
+		finalHeight          int32
+		nextCheckpointHeight int32
+	)
+
+	for _, header := range headers {
```

Comment:
See comment about `HeaderValidator[T]`. Then this just calls the method it exposes, and it doesn't need to house all this logic re the headers.